src/ReverseProxy/SessionAffinity/SessionAffinityMiddleware.cs

```diff
@@ -56,10 +56,14 @@ private async Task InvokeInternal(HttpContext context, IReverseProxyFeature prox
         var policy = _sessionAffinityPolicies.GetRequiredServiceById(config.Policy, SessionAffinityConstants.Policies.HashCookie);
         var affinityResult = await policy.FindAffinitizedDestinationsAsync(context, cluster, config, destinations, context.RequestAborted);
 
+        // Used for Distributed Tracing as part of Open Telemetry, null if there are no listeners
+        var activity = context.GetYarpActivity();
+
         switch (affinityResult.Status)
         {
             case AffinityStatus.OK:
                 proxyFeature.AvailableDestinations = affinityResult.Destinations!;
+                activity?.SetTag("Session Affinity", $"Destination selected by affinity policy {policy.Name}");
```

Comment:
Are the tag contents supposed to be messages like logs? That seems like it would take a lot of extra storage. You're also creating extra strings even if the activity doesn't exist.
```suggestion
                activity?.SetTag("SessionPolicySuccess", policy.Name);
                ...
                activity?.SetTag("SessionPolicyFailure", policy.Name);
```